core/controllers/admin.py

```diff
@@ -855,3 +855,15 @@ def post(self):
             self.render_json({})
         else:
             raise self.InvalidInputException('This app cannot send emails.')
+
+
+class UpdateUsernameHandler(base.BaseHandler):
+    """Handler for renaming usernames."""
+
+    @acl_decorators.can_access_admin_page
+    def put(self):
+        current_username = self.payload.get('current_username', None)
+        new_username = self.payload.get('new_username', None)
```

Comment:
Raise exceptions if any of the above value is None. Also, do we need to raise an exception if any of these values are not string before passing it to methods in user_services?

Reply:
Might also need to raise an exception if the new values don't obey the length limit.

Also, can you add that restriction to the signup page? Use 30 chars as the limit. (Assuming this doesn't break existing usernames -- please check.)

Reply:
@DubeySandeep, I've checked these scenarios while testing manually and all of them have been handled by the [require_valid_username](https://github.com/oppia/oppia/blob/146738b1dddd676630e0f919e8b5d2486508af52/core/domain/user_services.py#L262) method which gets called while setting the username. If any of the usernames (current or new) is None or contains non-alphanumeric characters or is longer than the threshold, the corresponding errors are raised.

![username_validation](https://user-images.githubusercontent.com/35144226/80708853-f800f400-8b09-11ea-9bba-2a5726f000b3.gif)

Also, @seanlip, the change in length limit is supposed to be done by #9051 which will be merged after the current usernames have been appropriately updated. But, I've added a check to ensure that the new usernames updated by the admin are not longer than 30 characters.